### PR TITLE
Set RCConstants

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: server/gokbg3/grails-app/controllers/org/gokb/rest/IdentifierController.groovy [hunk 1]
```diff
@@ -270,7 +271,7 @@ class IdentifierController {
   }
 
   private void fillTargetMap() {
-    RefdataValue.findAllByOwner(RefdataCategory.findByLabel('IdentifierNamespace.TargetType'))
+    RefdataValue.findAllByOwner(RefdataCategory.findByLabel(RCConstants.IDENTIFIER_NAMESPACE_TYPE))
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/controllers/org/gokb/rest/IdentifierController.groovy]: IdentifierNamespace.TargetType =? RCConstants.IDENTIFIER_NAMESPACE_TARGET_TYPE?

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy [hunk 2]
```diff
@@ -415,19 +416,19 @@ class PackageService {
         def has_book = pkg_obj.tipps.title.find { it.medium == rdv_book }
 
         if (has_db && !has_journal && !has_book) {
-          pkg_obj.contentType = RefdataCategory.lookup('Package.ContentType', 'Database')
+          pkg_obj.contentType = RefdataCategory.lookup('RCConstants.PACKAGE_CONTENT_TYPE', 'Database')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: usage of constants in strings? looks like typo ...

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_book.gsp [hunk 3]
```diff
@@ -47,22 +48,22 @@
   </dt>
   <dd>
     <gokb:xEditableRefData owner="${d}" field="reasonRetired"
-      config='TitleInstance.ReasonRetired' />
+      config='"${RCConstants.TITLEINSTANCE_REASON_RETIRED}"' />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_book.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_book.gsp [hunk 4]
```diff
@@ -196,7 +197,7 @@
         </dt>
         <dd>
           <gokb:xEditableRefData owner="${d}" field="OAStatus"
-            config='TitleInstance.OAStatus' />
+            config='"${RCConstants.TITLEINSTANCE_OA_STATUS}"' />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_book.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_database.gsp [hunk 5]
```diff
@@ -31,22 +32,22 @@
   </dt>
   <dd>
     <gokb:xEditableRefData owner="${d}" field="reasonRetired"
-      config='TitleInstance.ReasonRetired' />
+      config='"${RCConstants.TITLEINSTANCE_REASON_RETIRED}"' />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_database.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_database.gsp [hunk 6]
```diff
@@ -159,7 +160,7 @@
           </dt>
           <dd>
             <gokb:xEditableRefData owner="${d}" field="OAStatus"
-              config='TitleInstance.OAStatus' />
+              config='"${RCConstants.TITLEINSTANCE_OA_STATUS}"' />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_database.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_journal.gsp [hunk 7]
```diff
@@ -39,22 +40,22 @@
   </dt>
   <dd>
     <gokb:xEditableRefData owner="${d}" field="reasonRetired"
-                        config='TitleInstance.ReasonRetired'/>
+                        config='"${RCConstants.TITLEINSTANCE_REASON_RETIRED}"'/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_journal.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_journal.gsp [hunk 8]
```diff
@@ -189,7 +190,7 @@
         </dt>
         <dd>
           <gokb:xEditableRefData owner="${d}" field="OAStatus"
-                              config='TitleInstance.OAStatus'/>
+                              config='"${RCConstants.TITLEINSTANCE_OA_STATUS}"'/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_journal.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_org.gsp [hunk 9]
```diff
@@ -110,7 +111,7 @@
             <gokb:annotatedLabel owner="${d}" property="mission">Mission</gokb:annotatedLabel>
           </dt>
           <dd>
-            <gokb:xEditableRefData owner="${d}" field="mission" config='Org.Mission' />
+            <gokb:xEditableRefData owner="${d}" field="mission" config=${RCConstants.ORG_MISSION}" />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_org.gsp]: "" missing

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_othertitle.gsp [hunk 10]
```diff
@@ -27,30 +28,30 @@
...
+    <gokb:xEditableRefData owner="${d}" field="status" config="${RCConstants.KBCOMPONENT_STATUS}" />
   </dd>
 
   <dt>
     <gokb:annotatedLabel owner="${d}" property="reasonRetired">Status Reason</gokb:annotatedLabel>
   </dt>
   <dd>
     <gokb:xEditableRefData owner="${d}" field="reasonRetired"
-      config='TitleInstance.ReasonRetired' />
+      config='"${RCConstants.TITLEINSTANCE_REASON_RETIRED}"' />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_othertitle.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_othertitle.gsp [hunk 11]
```diff
@@ -123,15 +124,15 @@
...
+              config="${RCConstants.TITLEINSTANCE_MEDIUM}" />
           </dd>
 
           <dt>
             <gokb:annotatedLabel owner="${d}" property="OAStatus">OA Status</gokb:annotatedLabel>
           </dt>
           <dd>
             <gokb:xEditableRefData owner="${d}" field="OAStatus"
-              config='TitleInstance.OAStatus' />
+              config='"${RCConstants.TITLEINSTANCE_OA_STATUS}"' />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_othertitle.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_tipp.gsp [hunk 12]
```diff
@@ -119,7 +120,7 @@
         <gokb:annotatedLabel owner="${d}" property="publicationType">Publication Type</gokb:annotatedLabel>
     </dt>
     <dd>
-        <gokb:xEditableRefData owner="${d}" field="publicationType" config='TitleInstancePackagePlatform.PublicationType'/>
+        <gokb:xEditableRefData owner="${d}" field="publicationType" config='"${RCConstants.TIPP_PUBLICATION_TYPE}"'/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_tipp.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_tipp.gsp [hunk 13]
```diff
@@ -161,7 +162,7 @@
         <gokb:annotatedLabel owner="${d}" property="medium">Medium</gokb:annotatedLabel>
     </dt>
     <dd>
-        <gokb:xEditableRefData owner="${d}" field="medium" config='TitleInstancePackagePlatform.Medium'/>
+        <gokb:xEditableRefData owner="${d}" field="medium" config='"${RCConstants.TIPP_MEDIUM}"'/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_tipp.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_title.gsp [hunk 14]
```diff
@@ -39,22 +40,22 @@
   </dt>
   <dd>
     <gokb:xEditableRefData owner="${d}" field="reasonRetired"
-                        config='TitleInstance.ReasonRetired'/>
+                        config='"${RCConstants.TITLEINSTANCE_REASON_RETIRED}"'/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_title.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_title.gsp [hunk 15]
```diff
@@ -139,15 +140,15 @@
...
+                                config="${RCConstants.TITLEINSTANCE_MEDIUM}"/>
           </dd>
 
           <dt>
             <gokb:annotatedLabel owner="${d}" property="OAStatus">OA Status</gokb:annotatedLabel>
           </dt>
           <dd>
             <gokb:xEditableRefData owner="${d}" field="OAStatus"
-                                config='TitleInstance.OAStatus'/>
+                                config='"${RCConstants.TITLEINSTANCE_OA_STATUS}"'/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_title.gsp]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/secondTemplates/_simpleCombos.gsp [hunk 16]
```diff
@@ -50,7 +51,7 @@
     <input type="hidden" name="__context" value="${ctxoid}"/>
     <input type="hidden" name="__newObjectClass" value="org.gokb.cred.Combo"/>
     <input type="hidden" name="__recip" value="${recip}"/>
-    <input type="hidden" name="type" value="${org.gokb.cred.RefdataCategory.getOID('Combo.Type',d.getComboTypeValue(property))}"/>
+    <input type="hidden" name="type" value="${org.gokb.cred.RefdataCategory.getOID(de.wekb.helper.RCConstants.COMBO_TYPE,d.getComboTypeValue(property))}"/>
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/secondTemplates/_simpleCombos.gsp]: qualified call not needed

File: server/gokbg3/grails-app/controllers/org/gokb/ComponentController.groovy [hunk 17]
```diff
@@ -34,9 +35,9 @@ class ComponentController {
         KBComponent.withNewSession { session ->
           log.debug("fetching results for ${ns} ..")
 
-          RefdataValue status_deleted = RefdataCategory.lookup('KBComponent.Status', 'Deleted')
-          RefdataValue combo_type = RefdataCategory.lookup('Combo.Type', 'KBComponent.Ids')
-          RefdataValue combo_status = RefdataCategory.lookup('Combo.Status', 'Active')
+          RefdataValue status_deleted = RefdataCategory.lookup(RCConstants.KBCOMPONENT_STATUS, 'Deleted')
+          RefdataValue combo_type = RefdataCategory.lookup(RCConstants.COMBO_TYPE, 'KBComponent.Ids')
+          RefdataValue combo_status = RefdataCategory.lookup('RCConstants.COMBO_STATUS', 'Active')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/controllers/org/gokb/ComponentController.groovy]: superfluous ''

File: server/gokbg3/grails-app/domain/org/gokb/cred/TitleInstancePackagePlatform.groovy [hunk 18]
```diff
@@ -454,13 +456,13 @@ class TitleInstancePackagePlatform extends KBComponent {
...
     if (tipp_dto.medium) {
-      RefdataValue[] media = RefdataCategory.lookup("TitleInstance.Medium")
+      RefdataValue[] media = RefdataCategory.lookup(RCConstants.TITLEINSTANCE_MEDIUM)
       if (!media*.value.contains(tipp_dto.medium))
         errors.put('medium', [message: "unknown", baddata: tipp_dto.remove('medium')])
     }
 
     if (tipp_dto.publicationType) {
-      RefdataValue[] pubTypes = RefdataCategory.lookup("TitleInstancePackagePlatform.PublicationType")
+      RefdataValue[] pubTypes = RefdataCategory.lookup("${RCConstants.TIPP_PUBLICATION_TYPE}")
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/domain/org/gokb/cred/TitleInstancePackagePlatform.groovy]: unnecessary "${}"

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy [hunk 19]
```diff
@@ -415,19 +416,19 @@ class PackageService {
...
           result.db++
         }
         else if (has_journal && !has_db && !has_book) {
-          pkg_obj.contentType = RefdataCategory.lookup('Package.ContentType', 'Journal')
+          pkg_obj.contentType = RefdataCategory.lookup('RCConstants.PACKAGE_CONTENT_TYPE', 'Journal')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: superfluous ''

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy [hunk 20]
```diff
@@ -415,19 +416,19 @@ class PackageService {
...
           result.journal++
         }
         else if (has_book && !has_db && !has_journal) {
-          pkg_obj.contentType = RefdataCategory.lookup('Package.ContentType', 'Book')
+          pkg_obj.contentType = RefdataCategory.lookup('RCConstants.PACKAGE_CONTENT_TYPE', 'Book')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: superfluous ''

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy [hunk 21]
```diff
@@ -415,19 +416,19 @@ class PackageService {
...
           result.book++
         }
         else if (has_book && has_journal) {
-          pkg_obj.contentType = RefdataCategory.lookup('Package.ContentType', 'Mixed')
+          pkg_obj.contentType = RefdataCategory.lookup('RCConstants.PACKAGE_CONTENT_TYPE', 'Mixed')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: superfluous ''

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_license.gsp [hunk 22]
```diff
@@ -11,7 +12,7 @@
 			<gokb:annotatedLabel owner="${d}" property="licenseType">License Type</gokb:annotatedLabel>
 		</dt>
 		<dd>
-			<gokb:xEditableRefData owner="${d}" field="type" config='License.Type' />
+			<gokb:xEditableRefData owner="${d}" field="type" config="${de.wekb.helper.RCConstants.LICENSE_TYPE}" />
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/views/apptemplates/mainTemplates/_license.gsp]: qualified reference not needed

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_org.gsp (hunk 9 above)
Comment (croqueGrec09) [reply server/gokbg3/grails-app/views/apptemplates/mainTemplates/_org.gsp]: are still missing => config="${RCConstants.ORG_MISSION}"

File: server/gokbg3/grails-app/domain/org/gokb/cred/TitleInstancePackagePlatform.groovy [hunk 23]
```diff
@@ -674,8 +676,8 @@ class TitleInstancePackagePlatform extends KBComponent {
         changed |= com.k_int.ClassUtils.setDateIfPresent(tipp_dto.dateFirstInPrint, tipp, 'dateFirstInPrint')
         changed |= com.k_int.ClassUtils.setDateIfPresent(tipp_dto.dateFirstOnline, tipp, 'dateFirstOnline')
         changed |= com.k_int.ClassUtils.setDateIfPresent(tipp_dto.lastChangedExternal, tipp, 'lastChangedExternal')
-        changed |= com.k_int.ClassUtils.setRefdataIfPresent(tipp_dto.medium, tipp, 'medium', 'TitleInstance.Medium')
-        changed |= com.k_int.ClassUtils.setRefdataIfPresent(tipp_dto.publicationType, tipp, 'publicationType', 'TitleInstancePackagePlatform.PublicationType')
+        changed |= com.k_int.ClassUtils.setRefdataIfPresent(tipp_dto.medium, tipp, 'medium', RCConstants.TITLEINSTANCE_MEDIUM)
+        changed |= com.k_int.ClassUtils.setRefdataIfPresent(tipp_dto.publicationType, tipp, 'publicationType', 'RCConstants.TIPP_PUBLICATION_TYPE')
```
Comment (croqueGrec09) [inline server/gokbg3/grails-app/domain/org/gokb/cred/TitleInstancePackagePlatform.groovy]: superfluous ''

File: server/gokbg3/grails-app/controllers/org/gokb/rest/IdentifierController.groovy (hunk 1 above)
Comment (Moetez16) [reply server/gokbg3/grails-app/controllers/org/gokb/rest/IdentifierController.groovy]: Updated!

File: server/gokbg3/grails-app/services/org/gokb/PackageService.groovy (hunk 2 above)
Comment (Moetez16) [reply server/gokbg3/grails-app/services/org/gokb/PackageService.groovy]: Updated!

File: server/gokbg3/grails-app/views/apptemplates/mainTemplates/_book.gsp (hunk 3 above)
Comment (Moetez16) [reply server/gokbg3/grails-app/views/apptemplates/mainTemplates/_book.gsp]: Updated!

**Review by croqueGrec09 — APPROVED** [review]
approved

**Pull request merged**